benches/microbenches.rs

```diff
@@ -0,0 +1,482 @@
+
+            #[cfg(not(windows))]
+            assert_eq!(
+                nbtxt, 66277,
+                "Overall length (in bytes) of all text contents of ./tests/documents/sample_rss.xml"
+            );
+        });
+    });
+
+    group.bench_function("trim_text = true", |b| {
+        b.iter(|| {
+            let mut buf = Vec::new();
+            let mut r = Reader::from_reader(SAMPLE);
+            r.check_end_names(false)
+                .check_comments(false)
+                .trim_text(true);
+            let mut count = criterion::black_box(0);
+            let mut nbtxt = criterion::black_box(0);
+            loop {
+                match r.read_event(&mut buf) {
+                    Ok(Event::Start(_)) | Ok(Event::Empty(_)) => count += 1,
+                    Ok(Event::Text(ref e)) => nbtxt += e.unescaped().unwrap().len(),
+                    Ok(Event::Eof) => break,
+                    _ => (),
+                }
+                buf.clear();
+            }
+            assert_eq!(
+                count, 1550,
+                "Overall tag count in ./tests/documents/sample_rss.xml"
+            );
+
+            // Windows has \r\n instead of \n
+            #[cfg(windows)]
+            assert_eq!(
+                nbtxt, 50334,
+                "Overall length (in bytes) of all text contents of ./tests/documents/sample_rss.xml"
+            );
+
+            #[cfg(not(windows))]
+            assert_eq!(
+                nbtxt, 50261,
+                "Overall length (in bytes) of all text contents of ./tests/documents/sample_rss.xml"
+            );
+        });
+    });
+    group.finish();
+}
+
+/// Benchmarks, how fast individual event parsed
+fn one_event(c: &mut Criterion) {
+    let mut group = c.benchmark_group("One event");
+    group.bench_function("StartText", |b| {
+        let src = "Hello world!".repeat(512 / 12).into_bytes();
+        let mut buf = Vec::with_capacity(1024);
+        b.iter(|| {
+            let mut r = Reader::from_reader(src.as_ref());
+            let mut nbtxt = criterion::black_box(0);
+            r.check_end_names(false).check_comments(false);
+            match r.read_event(&mut buf) {
+                Ok(Event::StartText(e)) => nbtxt += e.len(),
+                something_else => panic!("Did not expect {:?}", something_else),
+            };
+
+            buf.clear();
+
+            assert_eq!(nbtxt, 504);
+        })
+    });
+
+    group.bench_function("Start", |b| {
+        let src = format!(r#"<hello target="{}">"#, "world".repeat(512 / 5)).into_bytes();
+        let mut buf = Vec::with_capacity(1024);
+        b.iter(|| {
+            let mut r = Reader::from_reader(src.as_ref());
+            let mut nbtxt = criterion::black_box(0);
+            r.check_end_names(false)
+                .check_comments(false)
+                .trim_text(true);
+            match r.read_event(&mut buf) {
+                Ok(Event::Start(ref e)) => nbtxt += e.name().as_ref().len(),
```

Comment:
`BytesStart` implements `Deref<[u8]>`, so why you replaced `e.len()` with `e.name().as_ref().len()`? Not that it was important...